compose/config.py

```diff
@@ -344,6 +346,63 @@ def env_vars_from_file(filename):
     return env
 
 
+def get_label_files(options, working_dir=None):
+    if 'label_file' not in options:
+        return {}
+
+    if working_dir is None:
+        raise Exception("No working_dir passed to label_file()")
+
+    label_files = options.get('label_file', [])
+    if not isinstance(label_files, list):
+        label_files = [label_files]
+
+    return [expand_path(working_dir, path) for path in label_files]
+
+
+def resolve_labels(service_dict, working_dir=None):
+    service_dict = service_dict.copy()
+
+    if 'labels' not in service_dict and 'label_file' not in service_dict:
+        return service_dict
+
+    labels = {}
+
+    if 'label_file' in service_dict:
+        for f in get_label_files(service_dict, working_dir=working_dir):
+            labels.update(label_vars_from_file(f))
+        del service_dict['label_file']
+
+    labels.update(parse_labels(service_dict.get('labels')))
+    labels = dict(resolve_label_var(k, v) for k, v in six.iteritems(labels))
+
+    service_dict['labels'] = labels
+    return service_dict
+
+
+def resolve_label_var(key, val):
+    if val is not None:
+        return key, val
+    else:
+        return key, ''
+
+
+def label_vars_from_file(filename):
+    """
+    Read in a line delimited file of labels variables.
+    """
+    if not os.path.exists(filename):
+        raise ConfigurationError("Couldn't find label file: %s" % filename)
+
+    labels = {}
+    for line in open(filename, 'r'):
+        line = line.strip()
+        if line and not line.startswith('#'):
+            k, v = split_label(line)
+            labels[k] = v
+    return labels
```

Comment:
A lot of this code is almost identical to  the `env_file` equivalent. We should just add a parameter to those functions and re-use the same code. I think that will make it easier to rebase.